docs/guides/understand_logs_and_metrics.md

```diff
@@ -0,0 +1,252 @@
+- `Argument size: 44.5 GB`: This is the memory needed to hold the inputs for your function. This typically includes the batch of data, parameter (master copy), and optimizer state (e.g., momentum).
+- `Output size: 44.5 GB`: This is the space required to store the results of the computation, such as the updated model weights and updated optimizer states.
+- `Temp size: 25.5 GB`: This is the "scratch space" memory. It's used for all the intermediate values created during the forward and backward passes that are discarded once the step is complete. This includes activation (forward pas), gradient (backward pass), and parameter (working copy, if mixed precision).
+- Q: Why it does not sum up? A: Some memory are shared (usually between argument and output).
+  - You might notice that the sum of the parts is greater than `70.0 GB (total)`: `44.5 GB (Argument) + 44.5 GB (Output) + 25.5 GB (Temp) = 114.5 GB`. The difference is due to a compiler optimization called memory aliasing. The compiler is smart enough to reuse memory blocks. The true calculation is `Total = Argument + Output + Temp - Aliased`. In our case, the compiler identified `44.5 GB (114.5 GB - 70.0 GB)` of memory that could be safely reused. Mostly likely, it reuses memory for `Argument` and `Output`.
+
+In addition, it shows temporary memory used on the host CPU. In this case, `Host temp size: 0.0 GB`, indicating that all the significant memory allocation happens on the accelerator device.
+
+
+### 3.2 Memory Snapshot 
+
+The previous section is a forecast of memory usage for entire training step, from static analysis of the compiled code. What about the actual memory usage? We now turn to the  snapshot of current usage right after the training state initialization, from a real-time check during execution. 
+
+To set stage for training, we first initialize the training state, which include parameter and optimizer states. At the [beginning](https://github.com/AI-Hypercomputer/maxtext/blob/f82ce194c490d668b14574a072a0a630c27bbd6e/MaxText/train.py#L695), the log shows a real-time snapshot of the [memory statistics](https://github.com/AI-Hypercomputer/maxtext/blob/f82ce194c490d668b14574a072a0a630c27bbd6e/MaxText/max_utils.py#L708-L717) on your TPU devices.
+
+```
+number parameters: 15.933 billion
+
+Memstats: After params initialized:
+	Using (GB) 44.6 / 95.74 (46.584500%) on TPU_0(process=0,(0,0,0,0))
+	Using (GB) 44.6 / 95.74 (46.584500%) on TPU_1(process=0,(1,0,0,0))
+	Using (GB) 44.6 / 95.74 (46.584500%) on TPU_2(process=0,(0,1,0,0))
+	Using (GB) 44.6 / 95.74 (46.584500%) on TPU_3(process=0,(1,1,0,0))
+```
+This log shows that each of the four TPUs has `95.74 GB` of available High Bandwidth Memory (HBM). The initial training state is evenly distributed devices, with each using the same amount of `44.6 GB`.
+
+We also note that `Argument size=44.5GB` from previous analysis is a close prediction of memory usage here.
```

Comment:
I didn't find too much explanation, but from example in JAX, it seems this total is peak memory. https://github.com/jax-ml/jax/issues/1491#issuecomment-1588287469

```
In [2]: jax.local_devices()[0].memory_stats()
Out[2]: 
{'bytes_in_use': 12288,
 'bytes_limit': 33014415360,
 'bytes_reservable_limit': 33014415360,
 'bytes_reserved': 0,
 'largest_alloc_size': 12288,
 'largest_free_block_bytes': 33014403072,
 'num_allocs': 1,
 'peak_bytes_in_use': 12288,
 'peak_bytes_reserved': 0}
 ```
 
Probably worth to mentioning?  Also, could we double check if this `bytes_in_use` from JAX should match with `Argument size` from compile? I am not clear on this.